src/effects/effectparameterslotbase.h

```diff
@@ -50,6 +51,10 @@ class EffectParameterSlotBase : public QObject {
 
     virtual void setParameter(double value) = 0;
 
+    virtual std::optional<double> neutralPointOnScale() const {
+        return std::nullopt;
+    }
```

Comment:
Then we'd have to override it for `EffectButtonParameterSlot`, just for the _possible_ case that the neutral point is requested for a button parameter?
That would otherwise return the unchanged default 0. Though, in that case (controlling a button parameter with a knob widget) a lot else wouldn't work quite as desired.
I don't mind..

Wdyt?

Reply:
> Then we'd have to override it for EffectButtonParameterSlot, just for the possible case that the neutral point is requested for a button parameter?

Well if that makes sense for a button parameter, we should make sure the value also makes sense. If it doesn't make sense (which is what you are suggesting, right?) then we can also let the return value be nonsensical. 

The advantage of implementing it in the base class is that we can avoid virtual dispatch. 

I think the primary issue here is that `neutralPointOnScale` only makes sense on knobs, but the existing code infrastructure requires it to be implemented on the base-class. Is that correct?

Reply:
Alright, implemented as you suggested. In the unlikely button case it'll return the default 0.